### PR TITLE
Multiple referrals: add instance-datum to suite XML

### PR DESCRIPTION
**Pull request by orangejenny**
## Technical Summary
https://dimagi-dev.atlassian.net/browse/USH-1784

## Feature Flag
USH: Allow selecting multiple cases from the case list

### Safety story
These changes are gated behind app configuration that isn't being used in production yet. Automated tests are sufficient to mitigate the risk of regressions.

### Automated test coverage

In PR.

### QA Plan

Not requesting QA.

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/apps/app_manager/suite_xml/xml_models.py [hunk 1]
```diff
@@ -540,12 +545,12 @@ class Entry(OrderedXmlObject, XmlObject):
     datums = NodeListField('session/datum', SessionDatum)
     queries = NodeListField('session/query', RemoteRequestQuery)
     session_children = NodeListField('session/*', _wrap_session_datums)
+    local_datums = NodeListField('session/*[self::datum or self::instance-datum]', _wrap_session_datums)
```
Comment (orangejenny) [inline corehq/apps/app_manager/suite_xml/xml_models.py:548]: I don't love adding this - it seems excessive to have the three individual attributes, then `session_children` to combine all three, then `local_datums` to combine just  datum and instance-datum - but I didn't find a better way to do it.

File: corehq/apps/app_manager/suite_xml/xml_models.py [hunk 2]
```diff
@@ -596,6 +601,7 @@ class RemoteRequestSession(OrderedXmlObject, XmlObject):
 
     queries = NodeListField('query', RemoteRequestQuery)
     data = NodeListField('datum', SessionDatum)
+    instance_data = NodeListField('instance-datum', InstanceDatum)
```
Comment (orangejenny) [inline corehq/apps/app_manager/suite_xml/xml_models.py:604]: I'm not sure if this is necessary/desirable. Nothing references it. But is _is_ part of the session, and I generally think the xml_models classes should be complete definitions.

File: corehq/apps/app_manager/suite_xml/xml_models.py (hunk 1 above)
Comment (stephherbers) [reply corehq/apps/app_manager/suite_xml/xml_models.py:548]: I can't think of a better way either, maybe @snopoke?

**Review by stephherbers — APPROVED** [review]
looks good, would still like to hear what Simon has to say given my so-so knowledge of this area

File: corehq/apps/app_manager/suite_xml/xml_models.py (hunk 1 above)
Comment (snopoke) [reply corehq/apps/app_manager/suite_xml/xml_models.py:548]: I guess there are 2 alternatives:

1. add an `instance_datums` field:   `for datum in entry.datums + entry.instance_datums:`
2. use `session_children` and do a type check: `if isinstance(child, (SessionDatum, InstanceDatum)`

I'm not sure that either is better than the current solution. (1) has the advantage of providing more granular access and being more similar to `RemoteRequestSession`

I'm also not sure about the name `local_datums`. I'm assuming that's in contrast to `remote_datums`? I might prefer `all_datums`.

**Review by snopoke — COMMENTED** [review]
How does this interact with form workflows?

File: corehq/apps/app_manager/suite_xml/xml_models.py (hunk 1 above)
Comment (orangejenny) [reply corehq/apps/app_manager/suite_xml/xml_models.py]: Updated the naming in d2178cb638f - it was to contrast with `remote_datums`, yes, but yeah, "local" doesn't really make sense.

> for datum in entry.datums + entry.instance_datums:

I originally did this and found it surprisingly challenging to join the two NodeListFields.

**Comment by orangejenny** [comment]
> How does this interact with form workflows?

Badly. I'm looking into that now, will do it in a separate ticket/PR. I' checking with delivery on what they need immediately, but generally I think a user should only be able to link to a multi-select form if they are coming from a multi-select form. Do you think that makes sense?

File: corehq/apps/app_manager/tests/test_suite_multi_select_case_list.py [hunk 3]
```diff
@@ -54,11 +54,11 @@ def test_multi_select_case_list(self):
                 </command>
                 <instance id="casedb" src="jr://instance/casedb"/>
                 <session>
-                  <datum id="case_id"
-                         nodeset="instance('casedb')/casedb/case[@case_type='person'][@status='open']"
-                         value="./@case_id"
-                         detail-select="m0_case_short"
-                         detail-confirm="m0_case_long"/>
+                  <instance-datum id="case_id"
```
Comment (shubham1g5) [inline corehq/apps/app_manager/tests/test_suite_multi_select_case_list.py:57]: @orangejenny the datum id should change to `selected_cases` for the instance('selected_cases') xpath to work

**Pull request merged**